Source/Core/Common/Network.cpp

```diff
@@ -170,6 +184,67 @@ u8 UDPHeader::IPProto() const
   return static_cast<u8>(IPPROTO_UDP);
 }
 
+ARPHeader::ARPHeader() = default;
+
+ARPHeader::ARPHeader(u32 from_ip, MACAddress from_mac, u32 to_ip, MACAddress to_mac)
+{
+  hardware_type = htons(BBA_HARDWARE_TYPE);
+  protocol_type = IPV4_HEADER_TYPE;
+  hardware_size = MAC_ADDRESS_SIZE;
+  protocol_size = IPV4_ADDR_LEN;
+  opcode = 0x200;
+  sender_ip = from_ip;
+  target_ip = to_ip;
+  targer_address = to_mac;
+  sender_address = from_mac;
+}
+
+u16 ARPHeader::Size() const
+{
+  return static_cast<u16>(SIZE);
+}
+
+DHCPBody::DHCPBody() = default;
+
+DHCPBody::DHCPBody(u32 transaction, MACAddress client_address, u32 new_ip, u32 serv_ip)
+{
+  transaction_id = transaction;
+  message_type = DHCPConst::MESSAGE_REPLY;
+  hardware_type = BBA_HARDWARE_TYPE;
+  hardware_addr = MAC_ADDRESS_SIZE;
+  client_mac = client_address;
+  your_ip = new_ip;
+  server_ip = serv_ip;
+}
+
+DHCPBody::DHCPBody(std::vector<u8> data)
+{
+  if (data.size() > SIZE)
+    return;
+
+  std::memcpy(this, data.data(), data.size());
```

Comment:
I'm not a fan of this approach using `memcpy` and the SIZE check might be too restrictive. Moreover, if what we are doing with the type-puning and strict-aliasing is undefined behaviour, I suspect that `this` might not be valid.

Reply:
It's valid only if the object is [trivially copyable](https://en.cppreference.com/w/cpp/named_req/TriviallyCopyable), so if you do this you need to add a `static_assert(std::is_trivially_copyable_v<DHCPBody>)` here.